vimeo/VimeoNetworking
# VIM-7487: Adds support for connected apps

### Ticket
[VIM-7487](https://vimean.atlassian.net/browse/VIM-7487)

### Pull Request Checklist

### Ticket Summary
Update VimeoNetworking to include support for the new Connected Apps endpoints.

### Implementation Summary

#### ConnectedApp
An object that contains all of the data related to a connection to a specific platform. However, not all of the properties relate to all platforms. Exceptions are noted in the documentation where applicable. Generally, `ConnectedApp` includes data related to when the connection was established, the platform it represents, and the scopes needed to interact with a particular feature. For now features are limited to live streaming and publishing to social.

#### Request+ConnectedApp
Adds convenience methods for constructing requests related to connected apps. This includes requesting all existing connected apps, creating a connection to a specific platform, and deleting an existing connection.

#### VIMConnection
Adds `VIMConnectionNameConnectedApps`, allowing a client to access the connected apps connection.

#### VIMInteraction
Expanded to support VIMUser interactions related to connected apps. You can now look for interactions on a user for each of the supported platforms (`facebook_connected_app`, for instance) to get information about whether the app is already connected, and what scopes are required for each of the supported features.

### Reviewer Tips
Testing this work requires access to the internal branch, `https://connected-apps-endpoints-api.ci.vimeows.com`. Furthermore, these features will only available to whitelisted apps for the foreseeable future once the changes are merged into the API.

### How to Test
Ensure unit tests pass. If testing against the CI branch ensure your client application is able to make the included requests and parse the results.
